### ISSUE TITLE
Usecase: Identity protection service

### ISSUE THREAD

**Issue by chrysn** [lake-wg/authz#26]
I'm not fully sure where this should go, but authz is probably a good starting point, because it is part of the solution I have in mind right now.

Consider two parties establishing a connection where both have sensitive identities (in terms of being tracked, of ownership, and both actively solicit connections continuously).
As an example, a service worker's phone might be constantly listening for beacons of installed equipment, which would report their latest logs, but neither do we want to allow tracking of the worker by identity or affiliation, nor do we want that arbitrary people driving through the city can map where they have equipment installed.
(I figure airtags probably do something similar, but it's easier when reporting your manufacturer just is 2-3 bits of information).

One way to solve this is to defer to a third party that is trusted with exactly one task: They can, when actively attacking (or just being used) read your identifying information, and promise to only relay that to other parties in your organization. The organization may run that on their own (in which case they'd probably pick LOC_W="" and take the risk of just seeing many decryption failures whenever they encounter devices of someone else who does the same), or they may use any global provider, where the number of providers should be smaller than the number of organizations using them in any area (maybe 5=len("GAMAM")...).

Using authz, this could be used like this:
(U = machine, V = phone, W = trustservice; as part of their enrollment, U has learned the public key of W and W knows the mapping from ID_U to CRED_U plus which account this is on, V has a secure channel with W associated with a customer account)
* U sends message 1 with its ID_U
* V sends the ENC_U_INFO to W
* W verifies that V is on the same customer account as U, and returns CRED_U as well as a voucher response.
  I'm not sure we even need to use the voucher here -- all that V needed to know that this is "one of its own", and knowing this it can now send its CRED_V (probably by value), and that will suffice to U
* V sends its CRED_U
* U recognizes the CRED_U and ... either sends a KID ID_CRED_V which suffices for this V, or if it turns out that V didn't contact W and just was generous with its CRED_V, maybe V sent something to ask for the full certificate.

Is that in scope as an application for authz? I think it'd work with authz as-is, and at most need some wording that "it is fine if the voucher response comes back OK without containing a voucher, V should then just continue like that". (Maybe the voucher is also just empty, and in the last step U takes that as the hint that even though V sent a credential, itself it can just send ID_CRED_I=b"").

**Comment by mcr** [comment]
> One way to solve this is to defer to a third party that is trusted with exactly one task: They can, when actively attacking (or just being used) read your identifying information, and promise to only relay that to other parties in your organization. The organization may run that on their own (in which case they'd probably pick LOC_W="" and take the risk of just seeing many decryption failures whenever they encounter devices of someone else who does the same), or they may use any global provider, where the number of providers should be smaller than the number of organizations using them in any area (maybe 4=len("GAMA")...).

I'm having difficulties understanding this paragraph.
I think that LOC_W="" means that V has to know the real value?

**Comment by chrysn** [comment]
Yes, LOC_W with some value could mean "You know whom to ask".

The point of the 3rd party is to enlarge the number of devices which one's device is indistinguishable from. Something is already gained from running such a service on your own (then devices are not trackable individually but any active attacker will see the domain part of their identity), a "you know whom to ask" works better at the cost of trial-and-error, and the moral equivalent of "check with cloudflare first" works more precisely and still just gives 2-3 bits of information (depending on whether there are 4 or 8 common providers).

**Comment by geonnave** [comment]
> One way to solve this ...

I also find it hard to understand this paragraph. It appears the main change is that W has a business logic commitment to not share ids with _other organizations_ which act as unintended trackers? If yes, I fail to see how to integrate it in the protocol.

> defer to a third party that is trusted with exactly one task: They can, when actively attacking (or just being used) read your identifying information

Who is attacking here? The trusted party? (that doesn't make sense to me)
Who can read the information?

> I'm not sure we even need to use the voucher here -- all that V needed to know that this is "one of its own"

I don't understand this part, as V already does not use the voucher ([Section 4.6.2.2](https://www.ietf.org/archive/id/draft-ietf-lake-authz-01.html#section-4.6.2.2)) -- the voucher's intended audience is U. 
V simply knows that the authorization has worked by receiving an "ok" answer from W (we need to actually define this, as tracked in #19).

**Comment by chrysn** [comment]
Sorry, I missed a few notifications here.

Apart from answering open questions (see next post), my question (or request to support) is that as far as ELA is concerned, *sening* a voucher should be optional (like, typically present, but allow that applications building on it do not need to convey any information from W to U).

**Comment by chrysn** [comment]
>  has a business logic commitment to not share ids with other organizations which could act as unintended trackers? If yes, I fail to see how to integrate it in the protocol.

No need to integrate that into the protocol, that is just policy.

> >  defer to a third party that is trusted with exactly one task: They can, when actively attacking (or just being used) read your identifying information
> Who is attacking here?

Let me rephrase: W is a trusted third party, but we do not have *much* trust in them. (About as much as you can trust any global organization that may be subject to any country's legislation). So *if* that third party were compromised in any way, all they can do is

* deny service to their users
* perform their duties as intended but monitor which parties V apparently saw which parties U (kind of necessary because that's what they're used to do, it's just that they should tell the *V*) or relay that information
* "wardrive", going around and correlate messages 1 with physical locations (and thus devices); unlike any untrusted party they can not only see *that* there is a party U around, but which user (eg. company) it belongs to

> I don't understand this part, as V already does not use the voucher ([Section 4.6.2.2](https://www.ietf.org/archive/id/draft-ietf-lake-authz-01.html#section-4.6.2.2)) -- the voucher's intended audience is U.
V simply knows that the authorization has worked by receiving an "ok" answer from W (we need to actually define this, as tracked in https://github.com/lake-wg/authz/issues/19).

The point here is that there can be cases where W's statement towards V is not "That peer is OK, here is a voucher to send on", but it is "That peer is OK, *and no need to send any voucher*" (because U is already equipped to process V's ID_CRED_V -- it's just that V is holding back on sending ID_CRED_V until W said that U is authorized to receive ID_CRED_V). This is not an error case, it's more a case of "yes that's null and that's fine, no need to send it".

I'm not so much worried about the V-W leg here -- IIUC, neither the cryptographic parts nor the U-V leg rely on that precise REST interface. But that U-V can go on after U sent an ELA EAD even if there is no response ELA EAD (because W knows the concrete application and can confidently know that there is nothing for U to process further) is something that the original EAD item description should enable.

---

So as I said, I hope this could be allowed in the document.

Speaking of allowing things: Anything wrong with having multiple voucher requests in the EAD1?

**Comment by geonnave** [comment]
Thanks for the clarifications. Some comments:

1. if the Voucher is there, and gets processed normally, is it doing any harm? is it preventing this use case to be implemented?
2. `Anything wrong with having multiple voucher requests?` I don't know what is the use case for that, but one technically could put more EAD_1 items in the EAD_1 field, and each could have a different LOC_W/ENC_U_INFO.
